### ISSUE TITLE
Performance issue with ContainerService Code

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#10101]
tldr; 
I know the old Velocity Service methods are ugly, but there is a reason for them.  They are very fast once they've been created, much faster than doing API calls in velocity. 

The introduction of the ContentStructureWebApi in the ContainerService render Container code has created a performance bottleneck:

Running 
```ab -n 10000 -c 25  http://127.0.0.1:8080/about-us/index```
I am getting  
```Requests per second:    58.54 [#/sec] (mean)```
which is not good. I took threadumps while running 

https://gist.github.com/wezell/32589b8927faf5719504461e67e3af0a

Most of the running threads are stuck here:
```at com.dotmarketing.viewtools.ContainerWebAPI.getStructureCode(ContainerWebAPI.java:49)```

If we do the work inline in the container code rather than calling a viewtool, I get:
```Requests per second:    256.90 [#/sec] (mean)```

**Comment by wezell** [comment]
Try this for a better diff:
https://github.com/dotCMS/core/commit/d6fbbd68a9013b133d9ba35835e0b540f5efeed6?w=1

**Comment by wezell** [comment]
Here are the resultant containers:

https://gist.github.com/wezell/f774c71efaa4e7cb50e21ee921bb45a5

**Comment by dsilvam** [comment]
Great catch @wezell !

**Comment by wezell** [comment]
Sucky, I do not think this is in 4.0 - it is in 4.1

**Comment by wezell** [comment]
In 4.0 Now
```
hot-fries:core will$ ab -n 10000 -c 25 http://127.0.0.1:8080/about-us/

Document Path:          /about-us/
Document Length:        23735 bytes

Concurrency Level:      25
Time taken for tests:   6.660 seconds
Complete requests:      360
Failed requests:        0
Total transferred:      8710200 bytes
HTML transferred:       8544600 bytes
Requests per second:    54.05 [#/sec] (mean)   <-----------------
Time per request:       462.497 [ms] (mean)
Time per request:       18.500 [ms] (mean, across all concurrent requests)
Transfer rate:          1277.19 [Kbytes/sec] received
```
With fix:
```
Server Software:        Apache-Coyote/1.1
Server Hostname:        127.0.0.1
Server Port:            8080

Document Path:          /about-us/
Document Length:        23735 bytes

Concurrency Level:      25
Time taken for tests:   4.457 seconds
Complete requests:      1000
Failed requests:        0
Total transferred:      24195000 bytes
HTML transferred:       23735000 bytes
Requests per second:    224.34 [#/sec] (mean)
Time per request:       111.436 [ms] (mean)
Time per request:       4.457 [ms] (mean, across all concurrent requests)
Transfer rate:          5300.80 [Kbytes/sec] received
```

**Comment by dsilvam** [comment]
Original PR: https://github.com/dotCMS/core/pull/10102

**Comment by rds4547** [comment]
can you confirm what was put into 3.6.1? I see this issue have fixed labels for both 3.6.1 and 4.0. Thank!

**Comment by dsilvam** [comment]
Yes, this is in 3.6.1 @rds4547
